cmd/soroban-rpc/internal/transactions/transactions.go

```diff
@@ -198,10 +199,33 @@ func (m *MemoryStore) GetTransaction(hash xdr.Hash) (Transaction, bool, StoreRan
 	if !ok {
 		return Transaction{}, false, storeRange
 	}
+
+	var txMeta xdr.TransactionMeta
+	err := txMeta.UnmarshalBinary(internalTx.meta)
+	if err != nil {
+		return Transaction{}, false, storeRange
+	}
+
+	txEvents, err := txMeta.GetDiagnosticEvents()
+	if err != nil {
+		return Transaction{}, false, storeRange
+	}
+
+	var events = make([][]byte, 0, len(txEvents))
+
```

Comment:
I think the style below is preferrable:

```suggestion
	events := make([][]byte, 0, len(txEvents))
```

var declarations are usually not combined with assignments